analyses/sample-distribution-analysis/04-ns-for-manuscript.Rmd

```diff
@@ -28,69 +29,127 @@ cell_lines_file <- file.path(root_dir, "analyses", "molecular-subtyping-HGG", "i
-tumor_bs <- hist %>%
-  filter(sample_type == "Tumor" & composition != "Derived Cell Line")
-nrow(tumor_bs)
+# Biospecimen assay counts by sample type and composition
+hist %>%
+  count(sample_type, composition, experimental_strategy)
+```
+## Calculate cell line Ns used for manuscript
 
-# Total number of unique cell line biospecimens (cell lines)
+```{r sample size - cell lines}
+# Total number of unique cell lines
 cellline_bs <- hist %>%
-  filter(sample_type == "Tumor" & composition == "Derived Cell Line")
-nrow(cellline_bs)
+  filter(sample_type == "Tumor" & composition == "Derived Cell Line") %>%
+  left_join(cell_lines, by = c("Kids_First_Participant_ID", "Kids_First_Biospecimen_ID", "sample_id")) %>%
+  select(Kids_First_Biospecimen_ID, sample_id, cell_line_composition, molecular_subtype) %>%
+  arrange(sample_id)
+ 
+cellline_bs_count <- cellline_bs %>%
+    count(sample_id, cell_line_composition)
+nrow(cellline_bs_count)
+
+
+# Total number of cell lines, including duplicates
+sum(cellline_bs_count$n)
+```
+## Calculate subtyped sample Ns used for manuscript
+
+```{r samples subtyped}
+# Total number of samples subtyped
+all_samples_subtyped <- hist %>%
+  filter(!is.na(pathology_diagnosis), 
+          !is.na(molecular_subtype), 
+          !grepl("To be classified", molecular_subtype)) %>% 
+   # Count sample_id and subtype combinations 
+  count(sample_id, molecular_subtype)
+nrow(all_samples_subtyped)
+
+# Total number of samples subtyped, not including cell lines
+tumors_not_lines_subtyped <- hist %>%
+  filter(composition != "Derived Cell Line",
+         !is.na(pathology_diagnosis), 
+         !is.na(molecular_subtype), 
+         !grepl("To be classified", molecular_subtype)) %>% 
+   # Count sample_id and subtype combinations 
+  count(sample_id, molecular_subtype)
+nrow(tumors_not_lines_subtyped)
+
+# percent of tumor samples subtyped
+(nrow(tumors_not_lines_subtyped)/length(n_tumors))*100
+
+# which have non-identical tumor/cell line subtypes?
+dup_samples <- all_samples_subtyped %>%
+  filter(duplicated(.[["sample_id"]])) %>%
+  pull(sample_id)
 
-# Total number of cell lines
-cell_lines %>%
-  select(sample_id, cell_line_composition) %>%
+hist %>%
+  filter(!is.na(molecular_subtype),
+         sample_id %in% dup_samples) %>%
+  count(sample_id, molecular_subtype, composition) %>%
+  arrange(sample_id, composition)
+
+# Do any cell line duplicates have different subtypes? No
+cellline_bs %>%
+  select(sample_id, molecular_subtype) %>%
   unique() %>%
-  nrow()
+  filter(duplicated(.[["sample_id"]])) %>%
+  pull(sample_id)
 
-# Total number of tumor biospecimens with WHO-aligned integrated diagnosis
-hist %>%
+# Total number of samples with WHO-aligned integrated diagnosis
+who_dx <- hist %>%
   filter(!is.na(integrated_diagnosis)) %>%
-  pull(Kids_First_Biospecimen_ID) %>%
-  unique() %>%
-  length()
+  count(sample_id, composition, molecular_subtype) 
+table(who_dx$composition)
```

Comment:
Yes, the reason is because some subtyping relied on just pathology free text, and in those cases, it's not considered "integrated" as it doesn't take into account the genomic data (eg neurocytoma which is subtyped based on primary site).